### ISSUE TITLE
Constraint engine?

### ISSUE THREAD

**Issue by bhouston** [elalish/manifold#77]
I wonder if it is possible to integrate a 2D constraint engine with this library?  That would be a very key step in moving towards proper CAD.

I wonder what FreeCAD uses for its 2D constraint system?  https://wiki.freecadweb.org/Sketcher_Workbench

**Comment by elalish** [comment]
Yes, although if what you want is traditional CAD, then you probably want NURBS and other feature-based flows. The idea of doing this in triangle meshes is that you can mix up the Booleans with sculpting and SDFs and generative design. A constraints engine would probably serve as an earlier stage: creating meshes and transforms for the inputs.
